mono/mini/method-to-ir.c

```diff
@@ -8237,6 +8237,9 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 
 			sp -= n;
 
+			if (!(cfg->method->wrapper_type && cfg->method->wrapper_type != MONO_WRAPPER_DYNAMIC_METHOD) && check_call_signature (cfg, fsig, sp))
```

Comment:
This wrapper check seems to recur. Can we make a function for it? I think tailcall checks it also.

Reply:
Kinda odd we're doing it for dynamic methods here, this doesn't feel like the sort of constraint they need to violate -  it's usually just visibility.